### PR TITLE
Remove redundant _device_t alias in torch.accelerator (#152952)

### PR DESCRIPTION
**Pull request by bobby404-K**
_device_t in torch/accelerator/_utils.py was just `Device` from torch.types re-exported under a different name. This continues the cleanup started by #161031 (which did the same for torch/cpu), moving on to the last remaining module still using the alias: torch.accelerator.

- torch/accelerator/_utils.py: drop the `as _device_t` re-export, import Device directly.
- torch/accelerator/__init__.py, memory.py, random.py: replace all _device_t annotations with Device, update imports accordingly.

No behavior change -- Device and _device_t were always the same type. torch.xpu, torch.mtia, and torch.cpu were already migrated by other contributors; this covers the one module (torch.accelerator) still using the old alias.

Fixes #152952

### CODE REVIEW

**Review by guangyey — DISMISSED** [review]
Thanks!

**Comment by guangyey** [comment]
@claude review

**Review by bobby404-K — COMMENTED** [review]
@pytorchbot merge

**Review by bobby404-K — COMMENTED** [review]
@pytorchbot merge

**Comment by bobby404-K** [comment]
@guangyey @ezyang 
Merge is blocked on two things that need maintainer action rather than a code change:

1. **BC Lint** is flagging the removal of `_device_t`. It's a private, underscore-prefixed alias that was never part of the public API (same pattern already accepted for torch.cpu in #161031), so this looks like a case for the `suppress-bc-linter` label rather than a real BC break.
2. **`ciflow/trunk`** hasn't run yet because the workflow is awaiting approval.

Could a maintainer apply the suppression label and approve the pending workflow run? Happy to make any other changes needed.

File: torch/accelerator/random.py [hunk 1]
```diff
@@ -1,7 +1,8 @@
 import torch
 from torch import Tensor
+from torch.types import Device as _device_t
 
-from ._utils import _device_t, _get_device_index
+from ._utils import _get_device_index
 
 
 def initial_seed(device: _device_t = None, /) -> int:
```
Comment (guangyey) [inline torch/accelerator/random.py]: Why did you change this code back?

**Comment by bobby404-K** [comment]
@guangyey Good catch — I missed this one during the cleanup. Fixed in the latest commit: _device_t → Device in both initial_seed and get_rng_state in random.py.

**Review by ezyang — APPROVED** [review]

**Comment by ezyang** [comment]
## mergedog handoff


> [!NOTE]
> This PR is being shepherded by [mergedog](https://github.com/ezyang/mergedog), an automation tool run by a maintainer. It watches CI and may push fixes; the PR author does not need to act unless a maintainer asks.

All CI is green (or skipped). Ready for human review and `@pytorchbot merge`.


Current PR head: `1285e7a83da3b8d38abe7bcc4a35d3c0a847a646`.
No mergedog-authored commits were pushed during this run.

No LLM was invoked during this run.

**Review by guangyey — APPROVED** [review]
